client-ui/src/app/model/operation/operation-property.models.ts

```diff
@@ -42,77 +42,61 @@ export namespace OperationProperty {
     json: ExtraPropertyValueJson[]
   }
 
-  export function decorateJsonProperty(propValue: ExtraPropertyValueJson): ExtraPropertyValueJson {
-    // make label human readable "camelCase" => "Camel Case"
-    const regex = new RegExp('([a-z])([A-Z])', 'g')
-    let label = propValue.label.replace(regex, '$1 $2')
-    label = label.substring(0, 1).toUpperCase() + label.substring(1)
-    return {
-      ...propValue,
-      label
-    }
-  }
-
-  export function decorateExpressionProperty(propValue: ExtraPropertyValue<Record<string, any>>,
-                                             attributesList: AttributeVM[]): ExtraPropertyValueExpression {
-    const jsonLikeProp = decorateJsonProperty(propValue)
-    return {
-      ...jsonLikeProp,
-      value: {
-        value: getText(propValue.value, attributesList),
-        rawValue: propValue.value
-      }
-    }
-  }
-
-  export function isExpresionProperty(propertyValue: Record<string, any>): boolean {
-    return propertyValue._typeHint && (propertyValue._typeHint as string).startsWith('expr.')
-  }
-
   export function parseExtraOptions(nativeProperties: NativeProperties, attributesList: AttributeVM[]): ExtraProperties {
-    const extraProperties: ExtraProperties = {
-      primitive: [],
-      expression: [],
-      json: []
-    }
-
-    return Object.keys(nativeProperties)
+    return Object.entries(nativeProperties)
+      .filter(([, v]) => v != null)
```

Comment:
having here filter instead of if statement inside of reduce you are iterating 2 times over whole array instead of 1. I don't think that this is a better approach.

Reply:
This is how you keep you reducer smaller and more focused it's primary function. This algorithm is dealing with usually very short arrays, so the code clarity should be a priority over tiny hypothetical speed gain.

Reply:
your reducer did not become significantly smaller, it is huge now. But you are iterating 2 times more now. 

I have nothing to say more here. I will mark that discussion as resolved.